test/stripe/account_test.rb

```diff
@@ -2,20 +2,35 @@
 
 module Stripe
   class AccountTest < Test::Unit::TestCase
-    should "be retrievable" do
-      resp = make_account({
-        :charges_enabled => false,
-        :details_submitted => false,
-        :email => "test+bindings@stripe.com",
-      })
-      @mock.expects(:get).
-        once.
-        with('https://api.stripe.com/v1/account', nil, nil).
-        returns(make_response(resp))
-      a = Stripe::Account.retrieve
-      assert_equal "test+bindings@stripe.com", a.email
-      assert !a.charges_enabled
-      assert !a.details_submitted
+    should "be retrievable with generated responses" do
+      without_legacy_stubs do
+        stub_api do
+          get "/v1/account" do
+            generated_response.merge!({
+              :charges_enabled => false,
+              :details_submitted => false,
+              :email => "test+bindings@stripe.com",
+            })
+          end
+        end
+
+        a = Stripe::Account.retrieve
+        assert_equal "test+bindings@stripe.com", a.email
+        assert !a.charges_enabled
+        assert !a.details_submitted
+      end
+    end
+
+    # This test is made to demonstrate that we could offer a potential route
+    # for very easy-to-build tests that provide a very basic check that a
+    # method hits an API endpoint and responds with something. The idea here is
+    # that it could help our situation where tests are hard to write
+    # (especially for new resources that need new `TestData`, and are therefore
+    # not written).
+    should "be retrievable with generated responses (easy edition)" do
+      without_legacy_stubs do
+        Stripe::Account.retrieve
```

Comment:
Good question. Yes indeed it would. I'll give you a few alternatives here in how we could handle this.

/cc @russelldavis For an opinion on this one as well.

First off, we could re-use the "traditional" test style where we inject values into a response and make sure those come back rather than whatever was generated:

``` ruby
    should "be retrievable with generated responses" do
      without_legacy_stubs do
        stub_api do
          get "/v1/account" do
            generated_response.merge!({
              :charges_enabled => false,
              :details_submitted => false,
              :email => "test+bindings@stripe.com",
            })
          end
        end

        a = Stripe::Account.retrieve
        assert_equal "test+bindings@stripe.com", a.email
        assert !a.charges_enabled
        assert !a.details_submitted
      end
    end
```

Secondly, we could take advantage of the fact that we're using closures here and do something simple like:

``` ruby
    should "be retrievable with generated responses" do
      without_legacy_stubs do
        received = false
        stub_api do
          get "/v1/account" do
            received = true
          end
        end

        Stripe::Account.retrieve
        assert call_received
      end
    end
```

Thirdly, I could built in a very basic mocking framework that could be called as `mock_api` instead of `stub_api`:

``` ruby
    should "be retrievable with generated responses" do
      without_legacy_stubs do
        mock = mock_api do
          get "/v1/account"
        end
        Stripe::Account.retrieve
        assert mock.satisfied?
      end
    end
```

The first option will be the easiest thing for now and will fall out naturally as we move all the tests over. We could add the second or third though so that we have a more forward-thinking direction here.